urdfenvs/urdfCommon/urdf_env.py

```diff
@@ -270,6 +274,10 @@ def set_walls(self, limits: list = [[-2, -2], [2, 2]]) -> None:
             [0, limits[1][1], 0.0],
             p.getQuaternionFromEuler([0, 0, np.pi / 2]),
         )
+ 
+        # refresh observation space for robots sensors
+        self.refresh_os()
```

Comment:
Why is this needed here? Isn't the function `get_obstacles()` called very often (potentially in every time step)? That would make the simulation very slow I assume.

Reply:
This is in the `set_walls()` function. Which adds obstacles and thus should the observation space be refreshed. 

The get_obstacles returns the list of obstacles and is here for completeness. But in the entire project, the function is never used.